assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,307 @@
+              var projects = projects.by("metadata.name");
+              for (var name in projects) {
+                $scope.buildFrom.projects.push(name);
+                $scope.pushTo.projects.push(name);
+              }
+              $scope.updateImageStreams($scope.options.pickedBuildFromNamespace, true, true);
+              $scope.updateImageStreams($scope.options.pickedPushToNamespace, false, true);
+            });
+
+
+            $scope.loaded = true;
+            // If we found the item successfully, watch for changes on it
+            watches.push(DataService.watchObject("buildconfigs", $routeParams.buildconfig, context, function(buildConfig, action) {
+              if (action === "DELETED") {
+                $scope.alerts["deleted"] = {
+                  type: "warning",
+                  message: "This build configuration has been deleted."
+                };
+              }
+              $scope.buildConfig = buildConfig;
+            }));
+          },
+          // failure
+          function(e) {
+            $scope.loaded = true;
+            $scope.alerts["load"] = {
+              type: "error",
+              message: "The build configuration details could not be loaded.",
+              details: "Reason: " + $filter('getErrorDetails')(e)
+            };
+          }
+        );
+      }));
+
+    // updateImageStreams creates/updates the list of imageStreams and ImageStreamTags from picked namespace. As parameter takes
+    // the picked namespace, buildFrom as a boolean that indicates whether the lists are for the builder image 'picker' or the 
+    // output imageStream 'picker'.
+    $scope.updateImageStreams = function(projectName, buildFrom, firstRun) {
+      DataService.list("imagestreams", {namespace: projectName}, function(imageStreams) {
+        if (buildFrom) {
+          if (!firstRun) {
+            $scope.options.pickedBuildFromImageStream = "";
+          }
+          $scope.buildFrom.imageStreams = [];
+          $scope.buildFrom.tags = {};
+        } else {
+          if (!firstRun) {
+            $scope.options.pickedPushToImageStream = "";
+          }
+          $scope.pushTo.imageStreams = [];
+          $scope.pushTo.tags = {};
+        }
+
+        var projectImageStreams = imageStreams.by("metadata.name");
+        angular.forEach(projectImageStreams, function(value, key) {
+          var tagList = [];
+          if (angular.isDefined(value.status.tags)) {
+            value.status.tags.forEach(function(item){ 
+              tagList.push(item["tag"]); 
+            });
+          }          
+
+          if (buildFrom) {
+            $scope.buildFrom.imageStreams.push(key);
+            $scope.buildFrom.tags[key] = tagList;
+          } else {
+            $scope.pushTo.imageStreams.push(key);
+            $scope.pushTo.tags[key] = tagList;
+          }
+        });
+      });
+
+    }
+
+    $scope.updateTags = function(imageStream, buildFrom) {
```

Comment:
Like above, I think this is better as two functions instead of taking a `buildFrom` param.